sklearn/preprocessing/_encoders.py

```diff
@@ -363,6 +486,201 @@ def _compute_drop_idx(self):
                              zip(self.drop, self.categories_)],
                             dtype=np.object)
 
+    def _infrequent_enabled(self):
+        """Infrequent category is enabled."""
+        if self.max_categories is not None and self.max_categories > 1:
```

Comment:
Why the second condition? i.e. why do we do validation here again?

Reply:
At one point I had this be a `@property` since it strictly depends on the parameters. At this point, we should define this once and use it throughout.